### ISSUE TITLE
Hostalias is not getting Effective

### ISSUE THREAD

**Issue by code-with-v** [aerokube/moon#319]
I tried hostAlias in the BrowserSet, but it's not getting effective. Please let me know if I did something wrong

 chrome:
      default: 99.0.4844.51-0
      hostAliases:
        - hostnames:
            - qa.local.dev
          ip: 192.168.140.200
      privileged: true
      repository: quay.io/browser/google-chrome-stable

**Comment by vania-pooh** [comment]
@vivek-singh-thridsummit check whether respective Chrome pod has exactly the same `hostAliases` section as yours.

**Comment by code-with-v** [comment]
> @vivek-singh-thridsummit check whether respective Chrome pod has exactly the same `hostAliases` section as yours.

No there isn't. Please check describe of the pod

```
PS C:\Windows\system32> kubectl describe pod chrome-99-0-4844-51-0-601f7312-672f-4665-ad71-14851007c8b9 -n moon
Name:         chrome-99-0-4844-51-0-601f7312-672f-4665-ad71-14851007c8b9
Namespace:    moon
Priority:     0
Node:         devops-worker-node-4/192.168.140.17
Start Time:   Mon, 18 Apr 2022 13:44:17 +0530
Labels:       browserName=chrome
              browserVersion=99.0.4844.51-0
              enableVNC=true
              enableVideo=false
              moon=browser
              quota=moon
              screenResolution=1920x1080
Annotations:  moon.aerokube.com/quota: moon
Status:       Running
IP:           10.244.4.232
IPs:
  IP:  10.244.4.232
Init Containers:
  ca-certs:
    Container ID:  docker://30792e7b5af0e386b9973532d8e45820a98408f7a96f22c677daad0d912b21fa
    Image:         quay.io/aerokube/ca-certs:2.1.1
    Image ID:      docker-pullable://quay.io/aerokube/ca-certs@sha256:8f73f08e977f42d71e725bcab1ff06abde7c1f1c2f33c80c6fa076618a455715
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      cat /etc/ssl/certs/ca-certificates.crt /certs/ca >/tmp/ca-certificates.crt
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Mon, 18 Apr 2022 13:44:18 +0530
      Finished:     Mon, 18 Apr 2022 13:44:18 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     200m
      memory:  32Mi
    Requests:
      cpu:        100m
      memory:     32Mi
    Environment:  <none>
    Mounts:
      /certs/ca from usergroup (rw,path="ca")
      /tmp from ca-certs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8wtkx (ro)
Containers:
  browser:
    Container ID:   docker://f365a147b37cb538383f49abdbf7cc4724e4d5f6b316956d06c6850c430cb9a3
    Image:          quay.io/browser/google-chrome-stable:99.0.4844.51-0
    Image ID:       docker-pullable://quay.io/browser/google-chrome-stable@sha256:1052548378f094cb475c9755bd3c9b4cb7f80f2979b1954768a7479b626c595d
    Port:           4444/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Mon, 18 Apr 2022 13:44:19 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  2Gi
    Requests:
      cpu:     500m
      memory:  2Gi
    Environment:
      DISPLAY:  127.0.0.1:0
    Mounts:
      /dev/shm from dshm (rw)
      /etc/group from usergroup (rw,path="group")
      /etc/passwd from usergroup (rw,path="passwd")
      /etc/ssl/certs from ca-certs (rw)
      /home/user from home (rw)
      /home/user/Downloads from downloads (rw)
      /tmp from tmp (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8wtkx (ro)
  defender:
    Container ID:  docker://cf58390a0a014302ca395ed11e140b13910bc3ba67ce1b4c635056bd5480f7c2
    Image:         quay.io/aerokube/defender:2.1.1
    Image ID:      docker-pullable://quay.io/aerokube/defender@sha256:fb8305624f599ceee76da84b6c2272d370a9916c081243ce53fc232ec582d5eb
    Port:          4545/TCP
    Host Port:     0/TCP
    Args:
      -moon-url
      http://moon.moon:4444/wd/hub
      -session-timeout
      5m
      -quota
      moon
    State:          Running
      Started:      Mon, 18 Apr 2022 13:44:19 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     250m
      memory:  128Mi
    Requests:
      cpu:     100m
      memory:  128Mi
    Environment:
      DISPLAY:  127.0.0.1:0
    Mounts:
      /dev/shm from dshm (rw)
      /etc/group from usergroup (rw,path="group")
      /etc/passwd from usergroup (rw,path="passwd")
      /etc/ssl/certs from ca-certs (rw)
      /home/user from home (rw)
      /home/user/Downloads from downloads (rw)
      /tmp from tmp (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8wtkx (ro)
  x-server:
    Container ID:   docker://3f30f8eceb92a7e1c9b2df96df6237356f6f171a1c755ad9ce43501db8bf3b43
    Image:          quay.io/aerokube/x-server:2.1.1
    Image ID:       docker-pullable://quay.io/aerokube/x-server@sha256:0583c50da6784ad3a34dcb9d0803feb84eb27ffe9b766418ef153a103018da8a
    Port:           6000/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Mon, 18 Apr 2022 13:44:20 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     250m
      memory:  128Mi
    Requests:
      cpu:     100m
      memory:  128Mi
    Environment:
      SCREEN_RESOLUTION:  1920x1080
    Mounts:
      /dev/shm from dshm (rw)
      /etc/group from usergroup (rw,path="group")
      /etc/passwd from usergroup (rw,path="passwd")
      /etc/ssl/certs from ca-certs (rw)
      /home/user from home (rw)
      /home/user/Downloads from downloads (rw)
      /tmp from tmp (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8wtkx (ro)
  vnc-server:
    Container ID:   docker://255ec33288e89a8f7fe7451ae51de59d897a1766e3fdfc7d013edd9c4bcd92f1
    Image:          quay.io/aerokube/vnc-server:2.1.1
    Image ID:       docker-pullable://quay.io/aerokube/vnc-server@sha256:3a771205bcea4c69d7dfab2cb1f60a411afe4c80e3fe92d31add5fd52e858823
    Port:           5900/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Mon, 18 Apr 2022 13:44:20 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     250m
      memory:  128Mi
    Requests:
      cpu:     100m
      memory:  128Mi
    Environment:
      DISPLAY:  127.0.0.1:0
    Mounts:
      /dev/shm from dshm (rw)
      /etc/group from usergroup (rw,path="group")
      /etc/passwd from usergroup (rw,path="passwd")
      /etc/ssl/certs from ca-certs (rw)
      /home/user from home (rw)
      /home/user/Downloads from downloads (rw)
      /tmp from tmp (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-8wtkx (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  dshm:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  <unset>
  usergroup:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      chrome-99-0-4844-51-0-601f7312-672f-4665-ad71-14851007c8b9
    Optional:  false
  tmp:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  <unset>
  home:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  <unset>
  downloads:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  ca-certs:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  <unset>
  kube-api-access-8wtkx:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  95s   default-scheduler  Successfully assigned moon/chrome-99-0-4844-51-0-601f7312-672f-4665-ad71-14851007c8b9 to devops-worker-node-4
  Normal  Pulled     94s   kubelet            Container image "quay.io/aerokube/ca-certs:2.1.1" already present on machine
  Normal  Started    94s   kubelet            Started container ca-certs
  Normal  Created    94s   kubelet            Created container ca-certs
  Normal  Pulled     93s   kubelet            Container image "quay.io/aerokube/x-server:2.1.1" already present on machine
  Normal  Pulled     93s   kubelet            Container image "quay.io/browser/google-chrome-stable:99.0.4844.51-0" already present on machine
  Normal  Created    93s   kubelet            Created container browser
  Normal  Started    93s   kubelet            Started container browser
  Normal  Pulled     93s   kubelet            Container image "quay.io/aerokube/defender:2.1.1" already present on machine
  Normal  Started    93s   kubelet            Started container defender
  Normal  Created    93s   kubelet            Created container defender
  Normal  Created    92s   kubelet            Created container x-server
  Normal  Started    92s   kubelet            Started container x-server
  Normal  Pulled     92s   kubelet            Container image "quay.io/aerokube/vnc-server:2.1.1" already present on machine
  Normal  Created    92s   kubelet            Created container vnc-server
  Normal  Started    92s   kubelet            Started container vnc-server
```

**Comment by vania-pooh** [comment]
@vivek-singh-thridsummit `hostAlias` flag in URL will work. Configuring this in `browserset` will be implemented in the next release.

**Comment by code-with-v** [comment]
> 

Could you please provide me the Syntax

**Comment by vania-pooh** [comment]
@vivek-singh-thridsummit the following capability should work:

```
"moon:options": {"hosts": ["qa.local.dev:192.168.140.200"]}
```

Note that in case of Java \ C# this should be added as Map \ Dictionary with an ArrayList inside and not string.

**Comment by vania-pooh** [comment]
Fixed in aerokube/moon2#112.
